Code/Framework/AzCore/Tests/Memory/PoolAllocatorTests.cpp

```diff
@@ -0,0 +1,125 @@
+            void* allocation = AZ::AllocatorInstance<TestPoolAllocator>::Get().Allocate(allocationSize, s_testPoolMinAllocationSize);
+            EXPECT_NE(nullptr, allocation);
+            EXPECT_LE(allocationSize, AZ::AllocatorInstance<TestPoolAllocator>::Get().AllocationSize(allocation));
+            allocations.emplace_back(allocation);
+        }
+
+        const size_t numberOfAllocations = allocations.size();
+        for (size_t i = 0; i < numberOfAllocations; ++i)
+        {
+            AZ::AllocatorInstance<TestPoolAllocator>::Get().DeAllocate(
+                allocations[i], testParameters.m_allocationSizes[i % testParameters.m_allocationSizes.size()]);
+        }
+
+    }
+
+    TEST_P(PoolAllocatorTestFixture, Allocate)
+    {
+        const PoolAllocatorTestParameters& testParameters = GetParam();
+        DoTestAllocations(testParameters);
+    }
+
+    static const AZStd::array<PoolAllocatorTestParameters, 2> s_smallInstancesParameters = {
+        PoolAllocatorTestParameters(s_smallAllocationSizes, 2), PoolAllocatorTestParameters(s_smallAllocationSizes, 100)
+    };
+    INSTANTIATE_TEST_CASE_P(SmallAllocation, PoolAllocatorTestFixture,
+        ::testing::ValuesIn(s_smallInstancesParameters));
+
+    static const AZStd::array<PoolAllocatorTestParameters, 2> s_bigInstancesParameters = {
+        PoolAllocatorTestParameters(s_bigAllocationSizes, 2), PoolAllocatorTestParameters(s_bigAllocationSizes, 100)
+    };
+    INSTANTIATE_TEST_CASE_P(BigAllocation, PoolAllocatorTestFixture,
+        ::testing::ValuesIn(s_bigInstancesParameters));
+
+    static const AZStd::array<PoolAllocatorTestParameters, 2> s_mixedInstancesParameters = {
+        PoolAllocatorTestParameters(s_mixedAllocationSizes, 2), PoolAllocatorTestParameters(s_mixedAllocationSizes, 100)
+    };
+    INSTANTIATE_TEST_CASE_P(MixedAllocation, PoolAllocatorTestFixture,
+        ::testing::ValuesIn(s_mixedInstancesParameters));
+
+    TEST_F(PoolAllocatorDeathTestFixture, PoolAllocator_AllocateLessThanMinAllocationSize_FailsToAllocate)
+    {
+        // Validate that the minimum allocation size of a custom PoolAllocator is respected
+        AZ::IAllocator::pointer result =
+            AZ::AllocatorInstance<TestPoolAllocator>::Get().Allocate(s_testPoolMinAllocationSize - 1, 1);
+        EXPECT_EQ(AZ::AllocatorInstance<TestPoolAllocator>::Get().NumAllocatedBytes(), s_testPoolMinAllocationSize);
+
+        // DeAllocate so the test doesn't leak
+        AZ::AllocatorInstance<TestPoolAllocator>::Get().DeAllocate(result);
+    }
+
+    TEST_F(PoolAllocatorDeathTestFixture, PoolAllocator_AllocateMoreThanMaxAllocationSize_FailsToAllocate)
+    {
+        // Validate that the maximum allocation size of a custom PoolAllocator is respected
+        AZ_TEST_START_ASSERTTEST;
+        AZ::IAllocator::pointer result =
+            AZ::AllocatorInstance<TestPoolAllocator>::Get().Allocate(s_testPoolMaxAllocationSize + 1, 1);
+        AZ_TEST_STOP_ASSERTTEST(1);
```

Comment:
`AZ_TEST_START_ASSERTTEST` and `AZ_TEST_STOP_ASSERTTEST` macros are [deprecated](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/UnitTest/UnitTest.h#L290-L301)

The `AZ_TEST_START_TRACE_SUPPRESSION` and `AZ_TEST_STOP_TRACE_SUPPRESSION` macros should be used